### ISSUE TITLE
Set noEmitOnError to true does not trigger live reloading

### ISSUE THREAD

**Issue by nightire** [typed-ember/ember-cli-typescript#1140]
### Please paste the output of `ember -v` here

ember-cli: 3.17.0
node: 12.16.1
os: darwin x64

### Please paste the output of `tsc -v` here

Version 3.6.5

### Please paste the version of `ember-cli-typescript` and `ember-cli-typescript-blueprints` here

ember-cli-typescript@4.0.0-alpha.1

### Please paste your `tsconfig.json` and `tslint.json` or `eslint.json` (if applicable) below

<details><summary><b>My tsconfig.json</b></summary><pre>
{
  "compilerOptions": {
    "allowJs": true,
    "alwaysStrict": true,
    "allowSyntheticDefaultImports": true,
    "experimentalDecorators": true,
    "inlineSourceMap": true,
    "inlineSources": true,
    "module": "ESNext",
    "moduleResolution": "node",
    "noEmit": true,
    "noEmitOnError": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "pretty": true,
    "strict": true,
    "target": "ESNext"
  }
}
</pre></details>

<details><summary><b>My tslint.json or eslint.json</b></summary><pre>N/A</pre></details>

### What are instructions we can follow to reproduce the issue?
```sh
ember new sample; cd ./sample # Create a new ember app
ember install ember-cli-typescript # Set up typescript support

1. `ember server`
2. Change any code to make a type error
```

##### Reproduction Case

N/A

### Now about that bug. What did you expect to see?

I expected to see the browser reload automatically and then shows the error message regarding the type error.

### What happened instead?

The browser didn't reload automatically, have to reload it manually to see the error message.

**Comment by dfreeman** [comment]
This may be related to https://github.com/ember-cli/ember-cli/issues/9074. We reject the returned promise from our [`postBuild`](https://ember-cli.com/api/classes/Addon.html#method_postBuild) if typechecking fails with `noEmitOnError` set in order to fail the entire build.

Up to recently this worked as expected, but about the time that linked issue was reported it started producing `UnhandledPromiseRejectionWarning`s. I'd guess that that's likely related to the reason livereload isn't being triggered, and the issue is likely in ember-cli, since it seems the `postBuild` promise isn't being handled?
